pkg/testcommon/testcommon.go

```diff
@@ -538,6 +538,45 @@ func GetLocalHTTPResponse(t *testing.T, rawurl string, params ...interface{}) (s
 	return bodyString, resp, lastErr
 }
 
+// GetLocalHTTPResponseWithBackoff calls GetLocalHTTPResponse with an external
+// retry/backoff strategy. It takes a number of attempts and an initialDelay
+// duration. params follow the same conventions as GetLocalHTTPResponse
+// (HTTPRequestOpts or int timeout). The inner call has MaxRetries forced to 1
+// to avoid nested retries.
+func GetLocalHTTPResponseWithBackoff(t *testing.T, rawurl string, attempts int, initialDelay time.Duration, params ...interface{}) (string, *http.Response, error) {
```

Comment:
I think this can be done more easily by adding more options to:
```go
// HTTPRequestOpts contains options for HTTP requests
type HTTPRequestOpts struct {
	// TimeoutSeconds is the number of seconds to wait for a response before timing out
	TimeoutSeconds int
	// MaxRetries is the number of times to retry the request
	MaxRetries int
}
```
and editing the original `time.Sleep` in `GetLocalHTTPResponse`

Reply:
And really... we don't need it any more at all, because the healthcheck was the root of the problem. Should we just pull this out?

Reply:
It's added in the tests, so it's not really important to me, let's pull it.